views/extras.tmpl

```diff
@@ -225,22 +227,33 @@
             $("#decode_header").text("Sent tx");
             $("#decoded_tx").text(evt);
         })
+
         var updateBlockData = function (event) {
             console.log("Received newblock message", event);
             var newBlock = JSON.parse(event);
-            var confirms = $('[data-confirmation-block-height]');
-            $.each(confirms, function(i, v) {
-                if (isNaN($(v).text())) {
-                    $(v).text("0");
-                }
-                var confirmations = newBlock.block.height - $(v).data('confirmation-block-height');
-                $(v).text("(" + confirmations + (confirmations > 1? " confirmations": " confirmation") + ")")
-            })
 
             // block summary data
             var b = newBlock.block;
             desktopNotifyNewBlock(b);
 
+            var currentTxHash = $('#spanTxID').data('tx-page-txid');
+            if(currentTxHash) { // we are on the tx page
+                // it is important to check that the current transaction is contained in the received block before any
+                // partial update. but since we are silently reloading the page, we will not have the issue of the view
+                // being wrongly updated that happens when the current transaction is not contained in the received block
+                Turbolinks.reload()
+            }else {
+                var confirms = $('[data-confirmation-block-height]');
+                $.each(confirms, function(i, v) {
+                    if (isNaN($(v).text())) {
+                        $(v).text("0");
+                    }
+                    var confirmations = newBlock.block.height - $(v).data('confirmation-block-height');
+                    // todo number of confirmations on the tx page should be handled seperately
+                    $(v).text("(" + confirmations + (confirmations > 1? " confirmations": " confirmation") + ")")
```

Comment:
In general as our javascript should be **idempotent**. Meaning the function can be called many times and the result is the same. 

So here for example, instead of incrementing, the function could just do current height - block height to get the # confirmations. 

We could use stimulus and make it a generic data target, similar to how age is handled.
So anywhere we need a dynamic # confirmations we can use set data-block-height and data-confirmations

Something like this:
https://github.com/decred/dcrdata/blob/master/views/block.tmpl#L126

With a stimulus controller that hooks into new block notifications and then does the delta calc and dom update
https://github.com/decred/dcrdata/blob/master/public/js/controllers/main.js#L35

Reply:
I'm all for keeping things simple and just reloading the page when determining accuracy is complex on the client side. Especially since Turbolinks makes that feel seamless to the end user.

How about keeping this decoupled?
We could use stimulus instead of jquery for a generic confirmations updater. 

And in situations when a user is viewing a block or transaction that is close to the tip ( eg. within 12 blocks ) the code governing those views can simply reload the page via Turbolinks new blocks come in as a straightforward way to ensure accuracy. 

I think it will really benefit us to avoid using imperative jQuery coding style when possible. I know it's super convenient and terse, but in the long run it inevitably leads to a spaghetti codebase. That is why I am suggesting to use stimulus for implementation. The logic of your approach seems good to me!